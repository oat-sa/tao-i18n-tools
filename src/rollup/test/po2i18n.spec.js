const poFileExampleContent = require('../../parsePoFile/test/parsePoFile.spec');
const po2I18n = require('../po2i18n');

describe('Rollup plugin API', () => {
    it('should exist', () => {
        expect(po2I18n).toBeDefined;
    });

    it('should be a function', () => {
        expect(po2I18n).toBeInstanceOf(Function);
    });

    it('skips non .po files', () => {
        expect(po2I18n({}).transform(poFileExampleContent, 'index.js')).not.toBeDefined();
    });
});

describe('correclty handles po files', () => {
    it('returns string with module containing translations as object and p11nRules as function', () => {
        const code = po2I18n({}).transform(poFileExampleContent, 'somefile.po').code;
        expect(code).toBeDefined();
        expect(code).toMatchSnapshot();
    });
});