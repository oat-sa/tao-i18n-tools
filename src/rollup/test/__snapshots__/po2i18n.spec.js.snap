// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correclty handles po files returns string with module containing translations as object and p11nRules as function 1`] = `"export default {"SINGULAR":"Singular message","CONTEXT.SINGULAR":"Singular message with context","PLURAL_MESSAGE":"Plural message","PLURAL_MESSAGES":["Plural message","Plural messages"]}; export const p11nRules = (n)=>(n != 1);"`;
